libraries/api-documenter/src/MarkdownPageRenderer.ts

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as fsx from 'fs-extra';
+import * as os from 'os';
+import * as path from 'path';
+
+import {
+  IDomPage,
+  DomElement
+} from './SimpleDom';
+
+import { BasePageRenderer } from './BasePageRenderer';
+
+/**
+ * Helper class used by MarkdownPageRenderer
+ */
+class SimpleWriter {
+  private _buffer: string = '';
```

Comment:
>_buffer [](start = 10, length = 7)

it is more performant to keep a list of strings and then join() them at the end

Reply:
This is incorrect for both Chrome and NodeJS environments.  Proof:

---
In reply to: [135167955](https://github.com/Microsoft/web-build-tools/pull/322#discussion_r135167955) [](ancestors = 135167955)

Reply:
```
function X() {
    var start = Date.now();
    for (var i = 0; i < 1000; ++i) {
        var buffer = '';
        for (var j = 0; j < 5000; ++j) {
            buffer += j.toString();
        }
    }
    var time = Date.now() - start;
    console.log('total time: ' + time);
}
function Y() {
    var start = Date.now();
    for (var i = 0; i < 1000; ++i) {
        var buffer = [];
        for (var j = 0; j < 5000; ++j) {
            buffer.push(j.toString());
        }
        var total = buffer.join();
    }
    var time = Date.now() - start;
    console.log('total time: ' + time);
}
console.log('-------------START');
X();
X();
X();
X();
console.log('-------------');
Y();
Y();
Y();
Y();
console.log('-------------');
X();
X();
X();
X();
console.log('-------------');
Y();
Y();
Y();
Y();
console.log('-------------END');
```

---
In reply to: [135374202](https://github.com/Microsoft/web-build-tools/pull/322#discussion_r135374202) [](ancestors = 135374202,135167955)